app/src/main/java/nodomain/freeyourgadget/gadgetbridge/service/devices/hplus/HPlusSupport.java

```diff
@@ -0,0 +1,881 @@
+            }
+        });
+        return this;
+    }
+
+    private HPlusSupport setSex(TransactionBuilder transaction) {
+        LOG.info("Attempting to set Sex...");
+        transaction.add(new ConditionalWriteAction(ctrlCharacteristic) {
+            @Override
+            protected byte[] checkCondition() {
+
+                byte value = HPlusCoordinator.getUserSex(getDevice().getAddress());
+                return new byte[]{
+                        HPlusConstants.COMMAND_SET_PREF_SEX,
+                        (byte) value
+                };
+            }
+        });
+        return this;
+    }
+
+
+    private HPlusSupport setGoal(TransactionBuilder transaction) {
+        LOG.info("Attempting to set Sex...");
+        transaction.add(new ConditionalWriteAction(ctrlCharacteristic) {
+            @Override
+            protected byte[] checkCondition() {
+
+                int value = HPlusCoordinator.getGoal(getDevice().getAddress());
+                return new byte[]{
+                        HPlusConstants.COMMAND_SET_PREF_GOAL,
+                        (byte) (value / 256),
+                        (byte) (value % 256)
+                };
+            }
+        });
+        return this;
+    }
+
+
+    private HPlusSupport setScreenTime(TransactionBuilder transaction) {
+        LOG.info("Attempting to set Screentime...");
+        transaction.add(new ConditionalWriteAction(ctrlCharacteristic) {
+            @Override
+            protected byte[] checkCondition() {
+
+                byte value = HPlusCoordinator.getScreenTime(getDevice().getAddress());
+                return new byte[]{
+                        HPlusConstants.COMMAND_SET_PREF_SCREENTIME,
+                        (byte) value
+                };
+            }
+        });
+        return this;
+    }
+
+    private HPlusSupport setAllDayHeart(TransactionBuilder transaction) {
+        LOG.info("Attempting to set All Day HR...");
+        transaction.add(new ConditionalWriteAction(ctrlCharacteristic) {
+            @Override
+            protected byte[] checkCondition() {
+
+                byte value = HPlusCoordinator.getAllDayHR(getDevice().getAddress());
+                return new byte[]{
+                        HPlusConstants.COMMAND_SET_PREF_ALLDAYHR,
+                        (byte) value
+                };
+            }
+        });
+        return this;
+    }
+
+
+    private HPlusSupport setAlarm(TransactionBuilder transaction) {
+        LOG.info("Attempting to set Alarm...");
+        return this;
+    }
+
+    private HPlusSupport setBlood(TransactionBuilder transaction) {
+        LOG.info("Attempting to set Blood...");
+        return this;
+    }
+
+
+    private HPlusSupport setFindMe(TransactionBuilder transaction) {
+        LOG.info("Attempting to set Findme...");
+        return this;
+    }
+
+    private HPlusSupport requestDeviceInfo(TransactionBuilder builder) {
+        LOG.debug("Requesting Device Info!");
```

Comment:
Does it perhaps implement the Bluetooth Device Info Profile? See class DeviceInfoProfile.